packages/teams-js/src/public/app.ts

```diff
@@ -806,6 +816,100 @@ export namespace app {
       resolve(sendAndHandleStatusAndReason('executeDeepLink', deepLink));
     });
   }
+
+  /**
+   * Namespace to opt into full-caching or delayed-termination of an app.
+   *
+   * @beta
+   */
+  export namespace lifecycle {
+    /**
+     * Register on resume handler function type
+     *
+     * @param context - Data structure to be used to pass the context to the app.
+     */
+    type registerOnResumeHandlerFunctionType = (context: ResumeContext) => void;
+    /**
+     * A function that is called when the app is ready to be suspended or terminated.
+     */
+    type ReadyToSuspendOrTerminateFunction = () => void;
+
+    /**
+     * Register before suspendOrTerminate handler function type
+     *
+     * @param readyToSuspendOrTerminate - A function that the handler is called with. Apps should call this function when they are ready.
+     * @returns void
+     */
+    type registerBeforeSuspendOrTerminateHandlerFunctionType = (
+      readyToSuspendOrTerminate: ReadyToSuspendOrTerminateFunction,
+    ) => void;
+
+    /**
+     * Registers a handler to be called before the page is suspended or terminatd.
+     *
+     * @param handler - The handler to invoke before the page is suspended or terminated. The page should
+     * invoke the readyToSuspendOrTerminate function provided to it once it's ready.
+     *
+     */
+    export function registerBeforeSuspendOrTerminateHandler(
+      handler: registerBeforeSuspendOrTerminateHandlerFunctionType,
+    ): void {
+      handler && ensureInitialized(runtime);
+      if (handler && !isSupported()) {
+        throw errorNotSupportedOnPlatform;
+      }
+      Handlers.registerBeforeSuspendOrTerminateHandler(handler);
+    }
+
+    /**
+     * Checks if app.lifecycle is supported by the host.
+     * @returns boolean to represent whether the lifecycle capability is supported
+     * @throws Error if {@linkcode app.initialize} has not successfully completed
+     *
+     * @beta
+     */
+    export function isSupported(): boolean {
+      return ensureInitialized(runtime) && runtime.supports.app && runtime.supports.app.lifecycle ? true : false;
+    }
+
+    /**
+     * Namespace to opt into full caching of an app
+     *
+     * @beta
+     */
+    export namespace caching {
```

Comment:
>caching [](http://example.com/codeflow?start=21&length=7)

Depending on how you feel about my comment below about the existence of this namespace all together, I wonder if this name would be better expressed as "resuming" or "suspension" or something, since you are modeling this as part of the app lifecycle (using appropriate terms like suspend and resume), and the developer-facing terminology moved away from "caching"

Reply:
Yep, it makes sense to change the name to resuming. I am going to wait on our decision on whether or not we need this sub capability before making the change. Thank you for the suggestion! 😊

Reply:
We are deleting this extra sub-sub capability and combining both handlers under `lifecycle`. Closing this comment for that reason